runtime/server/x86/frontend/fbank.h

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2017 Personal (Binbin Zhang)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef FRONTEND_FBANK_H_
+#define FRONTEND_FBANK_H_
+
+#include <assert.h>
+
+#include <vector>
+#include <limits>
+#include <random>
+#include <utility>
+
+#include "frontend/fft.h"
+
+#ifndef M_PI
+#define M_PI 3.1415926535897932384626433832795
+#endif
+#ifndef M_2PI
+#define M_2PI 6.283185307179586476925286766559005
+#endif
+
+namespace wenet {
+
+class Fbank {
+ public:
+  Fbank(int num_bins, int sample_rate, int frame_length, int frame_shift):
+      num_bins_(num_bins), sample_rate_(sample_rate),
+      frame_length_(frame_length), frame_shift_(frame_shift),
+      use_log_(true), remove_dc_offset_(true),
+      generator_(0), distribution_(0, 1.0), dither_(0.0) {
+    fft_points_ = UpperPowerOfTwo(frame_length_);
+    int num_fft_bins = fft_points_ / 2;
+    float fft_bin_width = static_cast<float>(sample_rate_) / fft_points_;
+    int low_freq = 20, high_freq = sample_rate_ / 2;
+    float mel_low_freq = MelScale(low_freq);
+    float mel_high_freq = MelScale(high_freq);
+    float mel_freq_delta = (mel_high_freq - mel_low_freq) / (num_bins+1);
+    bins_.resize(num_bins_);
+    center_freqs_.resize(num_bins_);
```

Comment:
Do you need to mention that the code is copied/modified from kaldi?

Reply:
okay